RFCs/0000-add-pdk-config.md

```diff
@@ -0,0 +1,292 @@
+# Add `pdk config` subcommand
+
+## Summary
+
+Introduce a PDK and module configuration subsystem with a new `pdk config` subcommand providing the user interface
+to read and write configuration keys and values.
+
+## Background & Assumptions
+
+Prior to the introduction of the Puppet Development Kit, module development involved the coordination of a set of
+distinct tools, rake tasks, etc. which each carried their own implicit and explicit mechanisms for customization and
+configuration.
+
+While the PDK has made a lot of strides in providing a unified invocation interface for these tools, thus far we have
+not addressed configuration side of the way module developers interact with these tools.
+
+To illustrate this, here is a survey of the current file-based configuration landscape for a single module:
+
+  - metadata.json
+  - .sync.yml
+  - .fixtures.yml
+  - .nodeset.yml
+  - .pdkignore
+  - .puppet-lint.rc
+  - .rspec
+  - .rubocop[\_todo].yml
+  - .yardopts
+  - Rakefile
+
+(Plus the inherent configuration and customization contained within the puppetlabs\_spec\_helper gem and it’s provided
+rake tasks, some of which the PDK invokes.)
+
+## Motivation & Goals
+
+- Standardize the way existing module development workflows are configured and customized for PDK users.
+
+- Reduce the number of different ways that module development workflows with PDK are configured and customized.
+
+- Resolve some inconsistencies and design challenges presented by existing PDK features. (See template-url/template-ref
+  design discussion and some issues we have had with invalid values cached in answers.json)
+
+- Improve the discoverability of module development configuration and customization. **Currently many of the avenues for
+  configuration are based around hidden files and their significance is not obvious to someone who is new to module
+  development.**
+
+## Proposed Implementation Design
+
+Implement a new pdk subcommand, `pdk config` which will offer the following actions:
```

Comment:
Quick question: Is the intention here to have a configuration command and config file structure similar to `git config`?

The command structure looks similar but the RFC does not contain a sample configuration file example. An example file in the RFC might help.